core/keyserver/keyserver.go

```diff
@@ -104,7 +99,7 @@ func (s *Server) getEntry(ctx context.Context, userID, appID string, firstTreeSi
 	})
 	if err != nil {
 		log.Printf("GetLeaves(): %v", err)
-		return nil, grpc.Errorf(codes.Internal, "Failed fetching map leaf")
+		return nil, grpc.Errorf(codes.Internal, "Failed fetching map leaf: %v", err)
```

Comment:
I thought we don't want to return the actual error back to the client, this is why we log it, right? We do this everywhere in this package and also in `core/mutation`. If we start returning the actual error to the client we should do it everywhere, but I think our current approach is fine.